src/trnpy/trnsys/simulation.py

```diff
@@ -115,6 +108,33 @@ def step_forward(self, steps: int = 1) -> bool:
 
         return done
 
+    def step_forward_with_values(self, steps: int = 1) -> StepForwardWithValuesReturn:
```

Comment:
Not for now, but we should expose this functionality as a proper generator.

Reply:
That's a great idea!  Tracked in #69.